src/main/kotlin/org/wycliffeassociates/otter/common/persistence/repositories/WorkbookRepository.kt

```diff
@@ -104,12 +113,27 @@ class WorkbookRepository(private val db: IDatabaseAccessors) : IWorkbookReposito
         val titleTextItem = textItem(title)
             ?: return null
 
-        return Resource(
+        val bodyComponent = body?.let {
+            textItem(it)?.let { bodyTextItem ->
+                Resource.Component(
```

Comment:
If the body text is null, should the body component be null or do we just return a null Resource?

Reply:
I don't know.

Reply:
why would you return a null resource? The resource could have a title?

Reply:
I assume that if the resource has an associated body Content, but the body content is empty, there is something wrong with the resource. In this case, do we still return the resource with a null body? In general, I'm not sure what to do when things happen that we don't expect. We seem to have a precedent for returning null objects in that case.

Reply:
> In general, I'm not sure what to do when things happen that we don't expect. We seem to have a precedent for returning null objects in that case.

Well this is a problem we need to address; we need to be very deliberate about why we use nullable types, and where.

My understanding is that a Resource is a pairing of Title and Body. If this assumption is true AND body is optional, then it would follow that we need to return resources that have a null body, because in that case the null is not a "null" as in error but null as in "optional." My assumption is that this is true because some things have headers with no body (which is why we have some resource card things with only one status indicator).

If I'm incorrect in this assumption, which use case is this exactly? If "Resource" is not meant to have an optional body, then the body field should not be nullable, and we should be handling some kind of error case here, not propagating nulls through the app's operation.

Reply:
You're correct, the body should be nullable since we may have headers with no body. I'm talking about the case where we are calling constructResource() to build a Resource, and the `body` parameter is not null BUT `textItem(body)` returns null. This means there is something wrong with the body Content object, i.e. either the `format` field of the Content object is incorrect or null, or the `text` field is null. In this case, we have found a faulty body Content, so I assume the entire Resource is corrupt. My question is then what do we do in this situation?

Reply:
Update: when `textItem` gets called, the `content` parameter will be a `Content` with a `contentType` of `TEXT`, `TITLE`, or `BODY`. In all of these cases, `content.format` and `content.text` should be non-null. HOWEVER, up until this point, UsfmProjectReader had been creating `Content` objects with null `format` and `text` fields. I have since changed that.

HOWEVER (x2), ParseUsfm does not appear to be working correctly, so for now, there are verse `Content` objects with null text fields. I am leaving a TODO here to fix this function once ParseUsfm is fixed. Even so, I have made the return type of `textItem` non-nullable since we do not want null TextItems propagating through the app.